apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -309,3 +321,106 @@ class FrameworkError {
   message: string;
   name: string;
 }
+
+function sanitizeControlValuesByOutputSchema(
+  controlValues: Record<string, unknown>,
+  type: StepTypeEnum
+): Record<string, unknown> {
+  const outputSchema = channelStepSchemas[type].output || actionStepSchemas[type].output;
+
+  if (!outputSchema || !controlValues) {
+    return controlValues;
+  }
+
+  const ajv = new Ajv({ allErrors: true });
+  addFormats(ajv);
+  const validate = ajv.compile(outputSchema);
+  const isValid = validate(controlValues);
+  const errors = validate.errors as null | ErrorObject[];
```

Comment:
I tested it locally and TS doesn't complain if we do `const { errors } = validate;`

Reply:
looks good, will do!